### DISCUSSION TITLE
Hide macOS tray icon

### DISCUSSION THREAD

**Discussion by heyscottyj** [ankurk91/google-chat-electron#49] [Q&A] [answered]
Love this app - thank you!

Is there a way to hide the menu bar icon on macOS? I believe there is a standard Electron function for this, but I am not a developer, and also didn't see an exposed option for this in preferences.

Thanks again!

ScottyJ

**Comment by ankurk91** [comment]
![google-chat-on-mac](https://user-images.githubusercontent.com/6111524/146318498-a1d72b8d-896a-439d-a805-fd731243e136.png)

I dont see any icon in mac os menu bar. Can you share your screenshot?

**Comment by heyscottyj** [comment]
![Screen Shot 2021-12-16 at 9 56 35 AM](https://user-images.githubusercontent.com/30265616/146414924-fb1135e8-d8ca-4de8-92d1-51a9a39717e1.png)

This is the one I am referring to (the green chat icon, which just sets a toggle to bring the window to front) - thanks for the reply!

**Comment by ankurk91** [comment, marked as answer]
This is feature that most chat apps has, this is called tray icon.

This helps the app to run in background.

**Reply by thefyrehead** [reply]
yes, that is correct. Most apps also offer it in black and white, and they also offer to hide it, use dock only, or both. Some even display notifications in the menubar, should you choose to want that. It's common macos culture to keep the menubar black and white and necessary icons only, hence grouping everything together in the control center. Just some self interested insight :-)

PS: The word Toggle is a bit confusing. It's used as a replacement "Show Window", Show, or Move to front, or unhide. And it's also used in beta to toggle features etc...

Regardless, this is helpful feedback, I am glad to trade a colored unnecessary menubar for this amazing little chat app I  enjoy using. I would pay 15 dollars to get notifications and little customisations like this.

**Reply by ankurk91** [reply]
@thefyrehead 

> Most apps also offer it in black and white, and they also offer to hide it, use dock only, or both.

This is a cross platform app, it requires a lot of time to make it work on those platforms.
I don't own a Mac, but i do have a VM to test the apps.

I can give an option to close the app completely (rather than keep it running in tray) when someone close it from window controls.

>  It's common macos culture to keep the menubar black and white

I am not a Photoshop guy, but i am happy to use black and white icons for tray icon if someone create them.

> The word Toggle is a bit confusing

The "Toggle" menu can be used to show/hide the app, if the app is hidden, this option will bring the app in front, if app is showing, this option will hide it.

**Reply by lpla** [reply]
> @thefyrehead
> 
> > Most apps also offer it in black and white, and they also offer to hide it, use dock only, or both.
> 
> This is a cross platform app, it requires a lot of time to make it work on those platforms. I don't own a Mac, but i do have a VM to test the apps.
> 
> I can give an option to close the app completely (rather than keep it running in tray) when someone close it from window controls.

In all chat apps I use in macOS (Skype, Rocket Chat, WhatsApp, Signal, Telegram) when you close the window, either using the red X button from the window or using CMD+w, apps keep running in the background. And you can check that in the dock icon because they have a dot below. But neither of them have a tray icon (or at least not one that cannot be disabled), because the restore/toggle/show window behaviour is duplicated in macOS if the tray icon is present, as you already have a dock pointing which apps are running in the background and able to reopen it. Could you try to reproduce this behaviour using a Virtual Machine of macOS?

**Reply by ankurk91** [reply]
Sorry I don't have time

**Comment by damianjamroz** [comment]
First of all, thank you for the effort in developing this app. It is fantastic!

I would like to support the idea to allow changing the menu icon to black and white, so it does not stand out among other apps.

<img width="611" alt="Screen Shot 2022-06-04 at 11 48 44 AM" src="https://user-images.githubusercontent.com/7937348/172013502-1d939d15-4f03-4db7-831a-ec8ab3983774.png">
 
I appreciate your consideration!

**Reply by BenMakesGames** [reply]
I've produced some black and white icons, in the following formats:
* SVG
* 48x48 PNG
* 128x128 PNG

@ankurk91, let me know if additional formats or resolutions are needed.

**SVG**
![chat-monochrome](https://user-images.githubusercontent.com/1587505/175330695-ff07eb4b-f7eb-41bc-b6e5-53f982daf625.svg)

**48x48 PNG**
![chat-monochrome-48x48](https://user-images.githubusercontent.com/1587505/175330699-37f24134-5f19-440a-a96c-55650a8c2298.png)

**128x128 PNG**
![chat-monochrome-128x128](https://user-images.githubusercontent.com/1587505/175330700-eba232cb-0c14-4ae0-9c68-1be54572d108.png)

**Reply by BenMakesGames** [reply]
it looks like the background became non-transparent as a result of attaching them to a github comment.

here are the source images, in a ZIP: [chat-monochrome.zip](https://github.com/ankurk91/google-chat-electron/files/8968493/chat-monochrome.zip)

**Reply by ankurk91** [reply]
There are 3 variants of the icon
https://github.com/ankurk91/google-chat-electron/tree/main/resources/icons

Png format should work, we will need to create icns file for mac os.
